File: NETCore/Cedrus/Controllers/V1/Models/ClassificationCaveatModel.cs
using Juniper.Cedrus.Entities;

namespace Juniper.Cedrus.Controllers.V1;

public class ClassificationCaveatModel
{
    public string TypeStamp => "classificationCaveat";
    public int? Id { get; }
    public string? Name { get; }
    public string? Description { get; }
    public ClassificationLevelModel Level { get; }

    public ClassificationCaveatModel(ClassificationCaveat? caveat)
    {
        Id = caveat?.Id;
        Name = caveat?.Name;
        Description = caveat?.Description;
        Level = new ClassificationLevelModel(caveat?.Level);
    }
}

File: JS/cedrus/src/models/ClassificationCaveatModel.ts
import { isString } from "@juniper-lib/util";
import { ClassificationLevelModel, isClassificationLevelModel } from "./ClassificationLevelModel";
import { IDOrName } from "./IDOrName";
import { INamed, isINamed } from "./INamed";
import { ITypeStamped, isTypeStamped } from "./ITypeStamped";

export interface ClassificationCaveatModel extends INamed, ITypeStamped<"classificationCaveat"> {
    description: string;
    level: ClassificationLevelModel;
}

export function isClassificationCaveatModel(obj: unknown): obj is ClassificationCaveatModel {
    return isTypeStamped("classificationCaveat", obj)
        && isINamed(obj)
        && "description" in obj
        && isString(obj.description)
        && "level" in obj
        && isClassificationLevelModel(obj.level);
}

export type SetClassificationCaveatInput = {
    classificationLevel: IDOrName;
    name: string;
    description: string;
}